pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -283,13 +293,75 @@ func (s *Reconciler) Update(ctx context.Context) error {
 		Message: machineCreationSucceedMessage,
 	})
 
-	vmState := v1beta1.VMState(*vm.ProvisioningState)
+	vmState := getVMState(vm)
 	s.scope.MachineStatus.VMID = vm.ID
 	s.scope.MachineStatus.VMState = &vmState
 
+	s.setMachineCloudProviderSpecifics(vm)
+
 	return nil
 }
 
+func getVMState(vm compute.VirtualMachine) v1beta1.VMState {
+	if vm.ProvisioningState == nil {
+		return ""
+	}
+
+	if *vm.ProvisioningState != "Succeeded" {
+		return v1beta1.VMState(*vm.ProvisioningState)
+	}
+
+	if vm.InstanceView == nil || vm.InstanceView.Statuses == nil {
+		return ""
+	}
+
+	for _, status := range *vm.InstanceView.Statuses {
+		if status.Code == nil || *status.Code == "ProvisioningState/succeeded" {
```

Comment:
For uniformity why not handle `status.Code == "ProvisioningState/succeeded"` as a `case`?